halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/component/v1/ServiceFactory.java

```diff
@@ -17,15 +17,17 @@
 package com.netflix.spinnaker.halyard.deploy.component.v1;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
 import retrofit.RestAdapter;
 import retrofit.client.OkClient;
 
-public class ServiceFactory<T extends ComponentService> {
+@Component
+public class ServiceFactory {
   @Autowired
   OkClient okClient;
 
-  public T createService(String endpoint, ComponentType componentType) {
-    Class<T> clazz = (Class<T>) componentType.getServiceClass();
+  public Object createService(String endpoint, ComponentType componentType) {
```

Comment:
Retrofit doesn't allow you to generate stubs for interfaces that extend other interfaces: https://github.com/square/retrofit/issues/504